core/templates/dev/head/services/csrf-token.service.spec.ts

```diff
@@ -21,57 +21,40 @@
 // needs to be imported explicitly.
 import $ from 'jquery';
 
-// TODO(#7222): Remove the following block of unnnecessary imports once
-// the code corresponding to the spec is upgraded to Angular 8.
-import { UpgradedServices } from 'services/UpgradedServices';
-// ^^^ This block is to be removed.
-
-require('services/csrf-token.service.ts');
+import { CsrfTokenService } from 'services/csrf-token.service';
 
 describe('Csrf Token Service', function() {
-  var $httpBackend = null;
-  var $q = null;
-  var $rootScope = null;
-  var CsrfTokenService = null;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.upgradedServices)) {
-      $provide.value(key, value);
-    }
-  }));
-  beforeEach(angular.mock.inject(function(
-      _$httpBackend_, _$q_, _$rootScope_, _CsrfTokenService_) {
-    $httpBackend = _$httpBackend_;
-    $q = _$q_;
-    $rootScope = _$rootScope_;
-    CsrfTokenService = _CsrfTokenService_;
-
-    this.scope = $rootScope.$new();
-
-    spyOn($, 'ajax').and.returnValue($q.resolve({token: 'sample-csrf-token'}));
-  }));
+  let csrfTokenService: CsrfTokenService = null;
+  beforeEach(() => {
+    csrfTokenService = new CsrfTokenService();
+
+    // This ts ignore is because returnValue is expecting a jqXHR<any>
+    // and wherease a Promise<{token: string}> is being provided.
+    // @ts-ignore
```

Comment:
I noticed this and wanted to ask about it. Shouldn't we find a way to fix it properly rather than using ts-ignore? /cc @bansalnitish @ankita240796

Reply:
@seanlip I tried some hacks to solve this but I don't think there is a clean way to do this.  Since the value it expects is a type of Jquery Object.

Reply:
Understood. I still think you should discuss it with the team though. At the very least an issue should be filed explaining all the things that don't work, but ideally a general solution can be found, since this is going to come up again in other files.

Reply:
I will look into this once and update here.

Reply:
Let's file a issue for this @Showtim3, also please update your PR.